### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by sebuliba256**
### All Submissions:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [ ] Have you added an explanation of what problem you are trying to solve with this PR?
* [ ] Have you added information on what your changes do and why you chose this as your solution?
* [ ] Have you written new tests for your changes?
* [ ] Does your submission pass tests?
* [ ] This project follows PEP8 style guide. Have you run your code against the 'flake8' linter?

### CODE REVIEW

File: docs/examples.rst [hunk 1]
```diff
@@ -277,8 +277,17 @@ select.
 Now start pyffd:
 
 .. code-block:: bash
-
-  # pyffd -f --loglevel=DEBUG -p /var/run/pyffd.pid mdx.fd
+  
+  # CACHE=-C
```
Comment (leifj) [inline docs/examples.rst]: -C means turn off cache - maybe be clear about that in the text
I like that you're using env variables to enhance readability!

File: docs/install.rst [hunk 2]
```diff
@@ -53,13 +50,32 @@ are developing pyFF or want to run multiple python-based applications in
...
 
 .. code-block:: bash
-
+  
+  # cd $HOME
   # apt-get install python-virtualenv
-  # mkdir -p /opt/pyff
-  # virtualenv /opt/pyff --no-site-packages
+  # mkdir python-pyff
+  # /opt/rh/rh-python36/root/usr/bin/virtualenv python-pyff --no-site-packages
```
Comment (leifj) [inline docs/install.rst]: Avoid referencing your particular install paths for python. Also https://stackoverflow.com/questions/23842713/using-python-3-in-virtualenv has an article on how virtualenv has changed in python3.

File: docs/install.rst [hunk 3]
```diff
@@ -53,13 +50,32 @@ are developing pyFF or want to run multiple python-based applications in
...
+Verifying
+----------
+
+To verify that python 3.6 is the default python in the pyFF environment run
+
+.. code-block:: bash
+
+  # python --version
+
+The result should be Python 3.6
```
Comment (leifj) [inline docs/install.rst]: 3.6 or later - I'm pretty sure we will be at 3.7 soon(ish)

File: docs/install.rst [hunk 4]
```diff
@@ -68,17 +84,32 @@ activating your virtualenv:
...
 
-  # pip install pyFF
+  # cd pyFF
+  # LANG=en_US.UTF-8 pip install -e .
 
 This will install a bunch of dependencies and compile bindings for both lxml, pyyaml
-aswell as pyXMLSecurity. This may take some time to complete. If there are no errors and if
-you have the *pyff* binary in your **$PATH** you should be done.
+as well as pyXMLSecurity. This may take some time to complete. If there are no errors and if
+you have the *python-pyff* binary in your **$PATH** you should be done.
```
Comment (leifj) [inline docs/install.rst]: the binary will still be python even if its in a virtualenv I think

File: docs/install.rst [hunk 5]
```diff
@@ -68,17 +84,32 @@ activating your virtualenv:
...
+
+.. code-block:: bash
+
+ # cd $HOME
+ # mkdir pyff-config
+ # cd pyff-config
+
+pyFF requires five files which you should copy into the pyff-config directory
+
+- A stylesheet in the *xsl* format
```
Comment (leifj) [inline docs/install.rst]: xslt stylesheets are not needed - only if you choose to use them in your pipeline and some of them are shipped with pyFF and can be referenced without copying them (eg tidy.xml). Refer to the xslt pipe documentation for details about how xslt files are located

File: docs/install.rst [hunk 6]
```diff
@@ -68,17 +84,32 @@ activating your virtualenv:
...
+- A metadata signer certficate
```
Comment (leifj) [inline docs/install.rst]: there is a script in the scripts-directory for generating a signing key pair for demo purposes

File: docs/install.rst [hunk 7]
```diff
@@ -68,17 +84,32 @@ activating your virtualenv:
...
+- A metadata signer key 
+- A pipeline file in the *yaml* format
+- A *xrd* file
```
Comment (leifj) [inline docs/install.rst]: xrd files are also not necessary - its just a way to represent a list of metadata URLs and their acoompanying signing certs but you could also do a load statement with multiple URL references... or a mix of xrd and URLs. There is also an (undocumented) way parser for so call MDSL files which are used in the eIDAS federation

File: docs/install.rst (hunk 6 above)
Comment (sebuliba256) [reply docs/install.rst]: If one uses the genkey script do they have to transfer the key pair to the pyff-config directory?

File: docs/using.rst [hunk 8]
```diff
@@ -1,48 +1,62 @@
...
-
-Documentation for each pipe is in the :py:mod:`pyff.pipes.builtins` Module. Also take a look at the :doc:`examples`.
+Using pyFF
+=============
+
+pyFF has two command-line tools: pyff and pyffd.
+
+.. code-block:: bash
+
+  # pyff --loglevel=INFO pipeline.fd [pipeline2.fd]
```
Comment (leifj) [inline docs/using.rst]: the code allows for loading multiple piplelines so the syntax should be something like [pipeline<N>.fd]+

**Pull request merged**